src/box/sql/select.c

```diff
@@ -1934,14 +1934,6 @@ generate_column_metadata(struct Parse *pParse, struct SrcList *pTabList,
 	if (pParse->colNamesSet)
```

Comment:
Thanks for the patch!

Could you please elaborate what was wrong here exactly. I can't see any problems in the old code.

Reply:
In old code memory was allocated according to number of variables, which can be different from number of variables uses, for example:
```
box.execute([[SELECT $1, $1, $1;]], {123})
```
Here we have only one variable, but we need three sets of metadata.

Reply:
Oh, I see, indeed.